aten/src/ATen/native/cuda/ScanKernels.cu

```diff
@@ -162,9 +168,8 @@ __host__ void scan_outer_dim_with_indices(const Tensor& self, Tensor& values, Te
   int num_irows = std::accumulate(sizes.begin() + dim + 1, sizes.end(), 1, std::multiplies<int>());
 
   dim3 threads(std::min(512, int(num_irows)));
-  int maxGridDim = at::cuda::getCurrentDeviceProperties()->maxGridSize[0];
+  int maxGridDim = at::cuda::getCurrentDeviceProperties()->maxGridSize[1];
```

Comment:
What is the reason for this?

Reply:
This was a bug, maxGridDim was used to set gridDim.y, yet it was querying the wrong device property (max gridDim.x, not gridDim.y). Since max gridDim.x is 2**31 and max gridDim.y is 65 K, it was easy to come up with invalid launch configuration (any tensor with outer dimension > 65K * 512 would do it).